hs-ontology-api-spec.yaml

```diff
@@ -339,6 +337,27 @@ paths:
                 type: array
                 items:
                   $ref: '#/components/schemas/SabCodeTerm'
+  /genes:
+    post:
```

Comment:
In REST API design, GET method should be used for retrieving details of the target resource. POST should be used to create data. We have the only POST call `/assayname` to accommodate a use case that was introduced long time ago, all other endpoints are GET. Since the UBKG neo4j is read-only, we are not supposed to create any new data with POST. Is there a way we can specify the ids in the query string rather than using a request body with json payload and convert it to a simple GET call?

Reply:
There is nothing preventing us from using GET instead of POST in the endpoint; however, the team that requested this endpoint explicitly requested that it be a POST request so that they could pass a large list of gene IDs.

Reply:
@AlanSimmons thanks for clarifying. In a strict RESTful API design manner, GET and POST should not be interchangeable, they serve different purposes. We have the implementation flexibility and can certainly use POST to accommodate a large payload if this is the intended use case. On the other hand, a GET call is cacheable while POST is not by default. If the portal-ui would need the requests to be cacheable for performance needs or even bookmarkable, I'll invite @lchoy to this discussion and decide.

Reply:
The portal could handle GET or POST. As I recall, the decision to use a POST request was in response to the need to support a list of IDs. With current mockups, we estimate the request may contain on the order of 50 IDs. While the ability to cache or bookmark the response has value, the portal does not require this at this time.

Reply:
@lchoy thanks for the input, we’ll proceed with POST.

Reply:
Wait--what? I just finished converting the endpoint to GET, and was ready to push another PR. Is POST now not evil?

Reply:
I commented out the POST content (route, SmartAPI). I can un-comment so that we have both a GET and a POST, or simply roll back changes and go with POST.

Reply:
@AlanSimmons my bad, I was reading @lchoy's message on my phone and probably misinterpreted. Since the portal-ui can handle both GET and POST, and only 50 IDs to be used, it's perfect to use GET to try to follow the best practices. And when I looked at the design document, it was also noted as GET for both `/proteins` and `/cell_type_detail`.

Reply:
@AlanSimmons I could have misinterpreted @lchoy's message when I read it on the phone. If you think sending the gene IDs in URL as query string will work perfectly fine, we should use GET to try to follow the best practices. On the other hand, if GET may pose limitations (more than 50 IDs or more complex structure rather than a simple list), we should use POST even thought POST is suboptimal for requests which do not modify a resource's state because it is not idempotent and limits or prevents client caching.

Really up to your judgement with future compatibility in mind. And there's no need to keep both in the code. In the end, the API serves the API consumers. One example is Elasticsearch query with POST, because it's not practical to specify complex query logics in the URL, Elasticsearch search API accepts POST with json body for read-only retrieval while supports GET with very simple query string parameters.

Reply:
Based on the above discussion and my understanding of the use case, we will use POST for this method. I have pushed a new commit of the API code (route, neo4j_logic, and SmartAPI YAML) that uses only POST. I have also updated the [specification](https://docs.google.com/document/d/1gW53B9YL7P3Qnm0sTvMuzz9DFvd4j80qDZFWtHwNavQ/edit?usp=sharing) to explain the use of POST.